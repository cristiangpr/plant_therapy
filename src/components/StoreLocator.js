
import React, { Component } from 'react'
import { Map, Info, Search } from 'react-store-locator'
import Navbar from './Navbar.js'
import myPin from '../myPin'
import mapStyle from '../mapStyle.json'
import locations from '../locations'

import clusterMarker from '../cluster-marker'


class StoreLocator extends Component {
	constructor(props) {
		super(props)

		this.state = {
			locations: locations,
			mapLoaded: false
		}

		this.getLocations = this.getLocations.bind(this)
	}

	getLocations(locations) {
		this.setState({ locations: locations })
		console.log(locations)
	}


	render() {
		const mapProps = {
			mapOptions: {
				styles: mapStyle,
				gestureHandling: `cooperative`,
				 fullscreenControl: false,
				 storeLocatorControl: false


			},
			onChange: this.getLocations,
			locations: this.state.locations,
			mapLoaded: () => {
				this.setState({ mapLoaded: true })
			},
			pin: { component: myPin },
			googleApiKey: process.env.REACT_APP_DEV_GOOGLE_MAPS_API

			// enableClusters: true,
			// cluster: {
			// 	component: clusterMarker,
			// 	radius: 100
			// }
		}

		return (
			<div className="App">

		<Navbar/>
      <div id="map">
				<Map {...mapProps}>
					{(location, closeLocation) => {
						return (
							<Info
								show={location.show}
								style={{

									height: '30px',
									backgroundColor: '#696969',
									width: '120px'
								}}
							>
								<div
									style={{
										textAlign: 'left',
										color: 'white',
										height: '22px', // Info height - padding - border to show border
										border: '1px solid white',
										padding: '3px',
										fontSize: '12px'
									}}
								>
									{location.name}
									<div
										style={{
											position: 'absolute',
											top: 3,
											right: 5,
											cursor: 'pointer',
											fontWeight: 800
										}}
										onClick={() => closeLocation(location.id)}
									>
										[x]
									</div>
								</div>
							</Info>
						)
					}}
				</Map>
</div>
				<h2>Locations In Window</h2>
				{this.state.locations.map(location => (
					<div
						key={location.id}
						style={{
							border: '1px solid #444',
							width: '25%',
							display: 'inline-block',
							marginTop: '5px',
							marginRight: '5px',
							padding: '10px',
							color: '#444',
							backgroundColor: "black"
						}}
					>
						<div>
							<h3
								style={{
									margin: '8px'
								}}
							>
								{location.name}
							</h3>
						</div>
						<div id ="distance">{location.distanceFromCenter} miles</div>
					</div>
				))}
			</div>
		)
	}
}

export default StoreLocator
