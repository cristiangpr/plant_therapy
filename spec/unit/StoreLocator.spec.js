import React from 'react';
import StoreLocator from "../../src/components/StoreLocator.js";

 import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';


describe(StoreLocator, function() {
  it('renders correctly', () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <StoreLocator/>
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
