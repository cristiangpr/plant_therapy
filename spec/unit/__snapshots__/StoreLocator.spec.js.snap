// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
<div
  className="App"
>
  <button
    id="mobile-nav-toggle"
    type="button"
  >
    <i
      className="fa fa-bars"
    />
  </button>
  <header
    id="header"
  >
    <div
      className="container-fluid"
    >
      <div
        className="pull-left"
        id="logo"
      >
        <h1>
          <a
            className="scrollto"
            href="/"
          >
            <img
              alt=""
              src="./img/logo.png"
              title=""
            />
          </a>
        </h1>
        <a
          href="#intro"
        />
      </div>
      <nav
        id="nav-menu-container"
      >
        <ul
          className="nav-menu sf-js-enabled sf-arrows"
        >
          <li
            className="menu-active"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="#about"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="#products"
            >
              Products
            </a>
          </li>
          <li>
            <a
              href="#blog"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="#footer"
            >
              Sign In
            </a>
          </li>
          <li>
            <a
              href="/farmsLanding"
              onClick={[Function]}
            >
              Farms
            </a>
          </li>
          <li>
            <a
              href="/distributorsLanding"
              onClick={[Function]}
            >
              Distributors
            </a>
          </li>
          <li>
            <a
              href="#contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <div
    id="map"
  >
    <div>
      Loading...
    </div>
  </div>
  <h2>
    Locations In Window
  </h2>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        First Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        Second Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        Third Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        Fourth Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        Fifth Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "black",
        "border": "1px solid #444",
        "color": "#444",
        "display": "inline-block",
        "marginRight": "5px",
        "marginTop": "5px",
        "padding": "10px",
        "width": "25%",
      }
    }
  >
    <div>
      <h3
        style={
          Object {
            "margin": "8px",
          }
        }
      >
        Sixth Marker
      </h3>
    </div>
    <div
      id="distance"
    >
       miles
    </div>
  </div>
</div>
`;
